src/kibana/plugins/settings/sections/objects/_objects.js

```diff
@@ -55,43 +69,103 @@ define(function (require) {
 
         $scope.edit = function (service, item) {
           var params = {
-            service: service.service,
+            service: service.serviceName,
             id: item.id
           };
 
           kbnUrl.change('/settings/objects/{{ service }}/{{ id }}', params);
         };
 
-        $scope.toggleDeleteBtn = function (service) {
-          $scope.deleteAllBtn = _.some(service.data, { checked: true});
+        $scope.bulkDelete = function () {
+          $scope.currentTab.service.delete(_.pluck($scope.selectedItems, 'id')).then(refreshData);
         };
 
-        $scope.bulkDelete = function () {
-          var serviceObj = _.find($scope.services, { title: $state.tab });
-          if (!serviceObj) return;
-          var service = $injector.get(serviceObj.service);
-          var ids = _(serviceObj.data)
-            .filter({ checked: true})
-            .pluck('id')
-            .value();
-          service.delete(ids).then(function (resp) {
-            serviceObj.data = _.filter(serviceObj.data, function (obj) {
-              return !obj.checked;
-            });
-            resetCheckBoxes();
+        $scope.bulkExport = function () {
+          var objs = $scope.selectedItems.map(_.partialRight(_.extend, {type: $scope.currentTab.type}));
+          retrieveAndExportDocs(objs);
+        };
+
+        $scope.exportAll = function () {
+          var objs = $scope.services.map(function (service) {
+            return service.data.map(_.partialRight(_.extend, {type: service.type}));
```

Comment:
Took me a minute to see what was going on here and on line 84. Maybe wrap this up like `transformHitsToExportBody()` or something.